docs/source/en/pipeline_tutorial.mdx

```diff
@@ -33,101 +33,202 @@ While each task has an associated [`pipeline`], it is simpler to use the general
+      {'text': 'HAVE', 'timestamp': (1.42, 1.58)},
+      {'text': 'A', 'timestamp': (1.66, 1.68)},
+      {'text': 'DREAM', 'timestamp': (1.76, 2.14)},
+      {'text': 'BUT', 'timestamp': (3.68, 3.8)},
+      {'text': 'ONE', 'timestamp': (3.94, 4.06)},
+      {'text': 'DAY', 'timestamp': (4.16, 4.3)},
+      {'text': 'THIS', 'timestamp': (6.36, 6.54)},
+      {'text': 'NATION', 'timestamp': (6.68, 7.1)},
+      {'text': 'WILL', 'timestamp': (7.32, 7.56)},
+      {'text': 'RISE', 'timestamp': (7.8, 8.26)},
+      {'text': 'UP', 'timestamp': (8.38, 8.48)},
+      {'text': 'AND', 'timestamp': (10.08, 10.18)},
+      {'text': 'LIVE', 'timestamp': (10.26, 10.48)},
+      {'text': 'OUT', 'timestamp': (10.58, 10.7)},
+      {'text': 'THE', 'timestamp': (10.82, 10.9)},
+      {'text': 'TRUE', 'timestamp': (10.98, 11.18)},
+      {'text': 'MEANING', 'timestamp': (11.26, 11.58)},
+      {'text': 'OF', 'timestamp': (11.66, 11.7)},
+      {'text': 'ITS', 'timestamp': (11.76, 11.88)},
+      {'text': 'CREED', 'timestamp': (12.0, 12.38)}
+]}
 ```
 
-Find an [audio classification](https://huggingface.co/models?pipeline_tag=audio-classification) model on the Model Hub for emotion recognition and load it in the [`pipeline`]:
+With that parameter, the model inferred the texts, but also outputted along **when** the various words were pronounced
+in the sentence.
+
+There are many parameters for all tasks, really check out the docs to find them.
+For instance this pipeline also handles `chunk_length_s` which is helpful to work on 
+really long audio files which the models cannot handle on their own. This is helpful when
+subtitling entire movies or hour-long videos.
+
+
+Cannot you find a parameter which would really help you out ?
+[Request it](https://github.com/huggingface/transformers/issues/new?assignees=&labels=feature&template=feature-request.yml)
+
+
+## Using pipelines on a dataset
+
+So you have a large dataset and want to run inference on it to get something like a
+metric for it, perfect !
+
+The easiest and more generic way we recomment is using an iterator.
 
 ```py
->>> from transformers import pipeline
+def data():
+    for i in range(1000):
+        yield f"My example {i}"
 
->>> audio_classifier = pipeline(
-...     task="audio-classification", model="ehcalabres/wav2vec2-lg-xlsr-en-speech-emotion-recognition"
-... )
+
+pipe = pipe(model="gpt2", device=0)
+generated_characters = 0
+for out in pipe(data()):
+    generated_characters += len(out["generated_text"])
 ```
 
-Pass the audio file to the [`pipeline`]:
+The iterator `data()` will yield each result. The pipeline will automatically
+recognize that the input is iterable, and will start fetching the data while
+it continues to process it on the GPU. This uses [DataLoader](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader) under the hood.
+This is important so you don't have to allocate memory for the whole dataset
+and you can feed the GPU as fast as possible.
```

Comment:
```suggestion
The iterator `data()` yields each result, and the pipeline automatically
recognizes the input is iterable and will start fetching the data while
it continues to process it on the GPU (this uses [DataLoader](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader) under the hood).
This is important because you don't have to allocate memory for the whole dataset
and you can feed the GPU as fast as possible.
```